### PR TITLE
[TM Ready] Monkey can have a little Donut - Return of Donutstation

### PR DESCRIPTION
**Pull request by williamcll**
# About The Pull Request
Brings back DonutStation, an old TG map now updated to bee standards.

## Why It's Good For The Game

Variety good, stop being addicted to metastation
## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
The full map

image
New Toxins Asteroid

image
Expanded holodeck, Added Exploration office and science shuttle.

image
Cleaner and updated Atmospherics

image
Expanded Perma
</details>

## Changelog

:cl:
add: DonutStation
add: DonutStation Exploration office
tweak: Updates Mining vendors
del: incompatible pianos
/:cl:

### CODE REVIEW

**Comment by williamcll** [comment]
TODO:
Add Exploration Office
Add Science Dock
Update Mining
Update Labour Shuttle
Update Pipe Layer
Update missing wiring

**Comment by polygoblyn** [comment]
Got any pics?

**Comment by nednaZ** [comment]
Linters are failing.

**Comment by williamcll** [comment]
> Linters are failing.

I don't know what's causing the issue, all the directories are in the right folder and there's no _maps in the _maps folder.

**Comment by nednaZ** [comment]
One thing I'll point out, you didn't create a branch for this.
I suggest creating new branches for every PR.

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -332,6 +332,13 @@
 	admin_notes = "It's big to spawn, it may or may not collide with the surrounding stuff on other maps that don't have a massive emergency docking area."
 	credit_cost = 8000
 
+/datum/map_template/shuttle/emergency/donut
+	suffix = "donut"
+	name = "Donut Station Emergency shuttle"
+	description = "A streamlined shuttle for a streamlined getaway from whatever hells going on."
+	admin_notes = "This shuttle hasn't updated since 2018, so there might be bugs."
```
Comment (nednaZ) [inline code/datums/shuttles.dm:339]: Has this shuttle been updated for this PR?

**Review by nednaZ — COMMENTED** [review]
So some immediate thoughts on the station design.

The vault is in a very weird place, placed in a maint tunnel directly next to security.
The vault also should not have turrets, no other map has that.

The supermatter, by being in the dead center, makes actually getting the SM shard objective far harder as there is no easy way to sneak in.

Permabrig is entirely impossible to escape via spacewalking.

Why is AI Upload in maintenance of all places?

The Warden's office needs Captain, Walter needs to be relocated to the Lawyer's Office.

Virology only has one lab setup, despite two spawns.

Exploration dock needs some kind of guideline to show new arrivals where it is.

Overall a functional, but INCREDIBLY weird map.
I don't get why some rooms are seemingly randomly sized, some are huge while others are tiny.

**Comment by williamcll** [comment]
Vault I can move it closer to the bridge

I can add another entrance like that of deltastation

Since when was perma possible to escape anyway? 

Blame the past people that designed this. Putting it in where the vault is now should be fine?

oversight, fill fix.

That I'm not sure how to do.

Mind you I didn't design this map, this was pulled from austation's upstream of the bee map from 2018

File: code/datums/shuttles.dm (hunk 1 above)
Comment (williamcll) [reply code/datums/shuttles.dm:339]: I added that note myself, walked around and I can't see any issues. Besides, this shuttle operates completely fine on NSV as well.

**Comment by williamcll** [comment]
Changed the following:
Added oxygen and firefighting locker around the primary hall
Updated solars APCs and naming
Added a green line guide from science entrance to exploration office
Added exploration vendor
Added exploration comms in telecomms.
Changed more fixed maints into random maints.
Added more entrances to Supermatter engine room.
Update Warden and Lawyer Pets.
Added one more setup for Virology.
Cleaned up atmos piping, no more pumps under doors.
Simplified Vault Access and removed turrets.
Changed AI Upload direction

**Review by nednaZ — COMMENTED** [review]
Being test merged.

**Comment by nednaZ** [comment]
Second testmerge being done.

**Review by nednaZ — APPROVED** [review]
Looks good to me.
I can't see any further issues that need immediate fixing, only long-term play will show issues that will need fixes

**Pull request merged**